machine_learning/Self_Organising_Maps.py

```diff
@@ -0,0 +1,62 @@
+import math
+
+
+class SOM:
+    # Function here computes the winning vector
+    # by Euclidean distance
+    def winner(self, weights, sample):
```

Comment:
These are type hints...
```suggestion
    def winner(self, weights: int, sample: int) --> list[int]:
```